### PR TITLE
Add a custom grid option for cesm-coupling

### LINKED ISSUE #343: Add a custom grid option to ctsm-coupling buildnml
As currently written the buildnml will fail if you tell mizuRoute you want to run for a grid that isn't explicitly defined in the list of valid grids. This is also explicit in the python and should be moved to a data file. We need a way to set a custom grid when running in CESM.

We are proposing that if you set LND_GRID to a custom option that it would then require the user to set the specific names of the files, but that you could then create a case.

We think this should be via setting the XML varialbe

LND_GRID

to "MIZU_USRDAT" to make it similar to the CLM_USRDAT grid that is used for CTSM.

Definition of done: Implement LND_GRID=MIZU_USRDAT option

### PR DESCRIPTION
**Pull request by ekluzek**
Add ability to add a custom grid option (with ROF_GRID=MIZU_USRDAT) so that you can do a custom grid setup.

Fixes #343

### CODE REVIEW

**Comment by nmizukami** [comment]
Hi @ekluzek, thanks! but...  could you pull-request to `cesm-coupling` branch instead of `main`??

File: cime_config/buildnml [hunk 1]
```diff
@@ -152,6 +152,16 @@ def _create_control_files(case, caseroot, srcroot, confdir, inst_string, infile,
        varname_segId = "seg_id"
        varname_downSegId = "Tosegment"
        varname_pfafCode = "pfaf"
+    elif (  config['rof_grid'] == "MIZU_USRDAT" ):
```
Comment (nmizukami) [inline cime_config/buildnml:155]: `ancil_dir` also needs to be set.  `ancil_dir` is directory where `fname_ntopOld` is located.   

I believe default `ancil_dir` is this. 
`<ancil_dir>                  /glade/p/cesmdata/cseg/inputdata/rof/mizuRoute/ancillary_data/     !  directory containing ancillary data (river network data)`

so add something like this?
`<ancil_dir>        "please_set_ancillary_directory_in_user_ctl_mizuroute"`

**Review by nmizukami — COMMENTED** [review]
I noticed one thing - `ancil_dir`.   I think user needs to set this in  `user_ctl_mizuroute`

Also, when you create the case,  is a user notified that `ancil_dir` and `fname_ntopOld` need to be set?

File: cime_config/buildnml (hunk 1 above)
Comment (ekluzek) [reply cime_config/buildnml:164]: OK, got it added in.

**Pull request merged**